src/WebGL/Browser.WebGL.fs

```diff
@@ -0,0 +1,909 @@
+    abstract TEXTURE3: float with get, set
+    abstract TEXTURE30: float with get, set
+    abstract TEXTURE31: float with get, set
+    abstract TEXTURE4: float with get, set
+    abstract TEXTURE5: float with get, set
+    abstract TEXTURE6: float with get, set
+    abstract TEXTURE7: float with get, set
+    abstract TEXTURE8: float with get, set
+    abstract TEXTURE9: float with get, set
+    abstract TEXTURE_2D: float with get, set
+    abstract TEXTURE_BINDING_2D: float with get, set
+    abstract TEXTURE_BINDING_CUBE_MAP: float with get, set
+    abstract TEXTURE_CUBE_MAP: float with get, set
+    abstract TEXTURE_CUBE_MAP_NEGATIVE_X: float with get, set
+    abstract TEXTURE_CUBE_MAP_NEGATIVE_Y: float with get, set
+    abstract TEXTURE_CUBE_MAP_NEGATIVE_Z: float with get, set
+    abstract TEXTURE_CUBE_MAP_POSITIVE_X: float with get, set
+    abstract TEXTURE_CUBE_MAP_POSITIVE_Y: float with get, set
+    abstract TEXTURE_CUBE_MAP_POSITIVE_Z: float with get, set
+    abstract TEXTURE_MAG_FILTER: float with get, set
+    abstract TEXTURE_MIN_FILTER: float with get, set
+    abstract TEXTURE_WRAP_S: float with get, set
+    abstract TEXTURE_WRAP_T: float with get, set
+    abstract TRIANGLES: float with get, set
+    abstract TRIANGLE_FAN: float with get, set
+    abstract TRIANGLE_STRIP: float with get, set
+    abstract UNPACK_ALIGNMENT: float with get, set
+    abstract UNPACK_COLORSPACE_CONVERSION_WEBGL: float with get, set
+    abstract UNPACK_FLIP_Y_WEBGL: float with get, set
+    abstract UNPACK_PREMULTIPLY_ALPHA_WEBGL: float with get, set
+    abstract UNSIGNED_BYTE: float with get, set
+    abstract UNSIGNED_INT: float with get, set
+    abstract UNSIGNED_SHORT: float with get, set
+    abstract UNSIGNED_SHORT_4_4_4_4: float with get, set
+    abstract UNSIGNED_SHORT_5_5_5_1: float with get, set
+    abstract UNSIGNED_SHORT_5_6_5: float with get, set
+    abstract VALIDATE_STATUS: float with get, set
+    abstract VENDOR: float with get, set
+    abstract VERSION: float with get, set
+    abstract VERTEX_ATTRIB_ARRAY_BUFFER_BINDING: float with get, set
+    abstract VERTEX_ATTRIB_ARRAY_ENABLED: float with get, set
+    abstract VERTEX_ATTRIB_ARRAY_NORMALIZED: float with get, set
+    abstract VERTEX_ATTRIB_ARRAY_POINTER: float with get, set
+    abstract VERTEX_ATTRIB_ARRAY_SIZE: float with get, set
+    abstract VERTEX_ATTRIB_ARRAY_STRIDE: float with get, set
+    abstract VERTEX_ATTRIB_ARRAY_TYPE: float with get, set
+    abstract VERTEX_SHADER: float with get, set
+    abstract VIEWPORT: float with get, set
+    abstract ZERO: float with get, set
+    [<Emit("new $0($1...)")>] abstract Create: unit -> WebGLRenderingContext
+
+namespace Browser
+
+open Browser.Types
+open Fable.Core
+
+module WebGL =
+
+    type HTMLCanvasElement with
+        [<Emit("$0.getContext('experimental-webgl')")>]
+        member __.``getContext_experimental-webgl``: unit -> WebGLRenderingContext = failwith ""
```

Comment:
As we're emitting JS anyways we could try to avoid the hyphen (either with underscore or a capital W letter) to remove the quotes.

Reply:
This syntax will confuse the compiler because it will use the content of emit when accessing the property and then apply the unit argument: `el.getContext('experimental-webgl')()`. You can avoid that by writing it as a method instead:

```fsharp
type HTMLCanvasElement with
        [<Emit("$0.getContext('experimental-webgl')")>]
        member __.getContext_experimental_webgl(): WebGLRenderingContext = jsNative
```